libs/payments/cart/src/lib/cart.service.spec.ts

```diff
@@ -478,6 +479,68 @@ describe('CartService', () => {
         mockUpdateCart
       );
     });
+
+    it('updates cart with coupon code', async () => {
+      const mockCart = ResultCartFactory();
+      const mockUpdateCart = UpdateCartFactory({
+        couponCode: faker.word.noun(),
+      });
+      const mockPrice = StripePriceFactory();
+
+      jest.spyOn(cartManager, 'fetchCartById').mockResolvedValue(mockCart);
+      jest
+        .spyOn(productConfigurationManager, 'retrieveStripePrice')
+        .mockResolvedValue(mockPrice);
+      jest
+        .spyOn(promotionCodeManager, 'assertValidPromotionCodeNameForPrice')
+        .mockResolvedValue(undefined);
+
+      jest.spyOn(cartManager, 'updateFreshCart').mockResolvedValue();
+
+      await cartService.updateCart(
+        mockCart.id,
+        mockCart.version,
+        mockUpdateCart
+      );
+
+      expect(cartManager.updateFreshCart).toHaveBeenCalledWith(
+        mockCart.id,
+        mockCart.version,
+        mockUpdateCart
+      );
+    });
+
+    it('throws if coupon is not valid', async () => {
+      const mockCouponCode = faker.word.noun();
+      const mockPrice = StripePriceFactory();
+      const mockCart = ResultCartFactory({
+        couponCode: mockCouponCode,
+      });
+      const mockUpdateCart = UpdateCartFactory({
+        couponCode: mockCouponCode,
+      });
+
+      jest.spyOn(cartManager, 'fetchCartById').mockResolvedValue(mockCart);
+      jest
+        .spyOn(productConfigurationManager, 'retrieveStripePrice')
+        .mockResolvedValue(mockPrice);
```

Comment:
Much of this seems to be a duplicate of what's found in the previous test. Could the duplicate portions be moved into the describe block instead?

This leaves the test a little lighter and easier to parse.